src/main/java/org/skife/jdbi/v2/unstable/Objects.java

```diff
@@ -0,0 +1,27 @@
+package org.skife.jdbi.v2.unstable;
+
+import java.lang.reflect.Array;
+import java.util.Collection;
+
+final class Objects {
+    private Objects() {}
+
+    static Object[] toArray(final Object obj) {
+        if (obj instanceof Collection) {
+            return ((Collection) obj).toArray();
+        }
+
+        if (obj.getClass().isArray()) {
+            final int length = Array.getLength(obj);
+            final Object[] out = new Object[length];
```

Comment:
You only need to copy the array if `obj.getClass().getComponentType().isPrimitive()`. Otherwise you can just cast to `Object[]`